checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -354,9 +606,342 @@ protected boolean skipContractCheck(Tree tree, FlowExpressions.Receiver expr, Fl
                 }
             }
         } catch (FlowExpressionParseException e) {
-            // errors are reported at declaration site
+            checker.report(e.getResult(), tree);
         }
 
         return false;
     }
+
+    @Override
+    public boolean isValidUse(AnnotatedDeclaredType declarationType,
+            AnnotatedDeclaredType useType, Tree tree) {
+        declarationType.replaceAnnotation(GUARDEDBY);
```

Comment:
Thanks for the explanation.  This is a (known) limitation/bug.  The right thing would be to change to default of class declarations to `@GuardedByUnknown`, but this would cause warnings in unexpected places. So, instead, this method should just skip this check by always returning true, which is what this code is doing for the GuardedBy hierarchy.   If you don't need this check for the LockHeld hierarchy, then I would change this method to just return true.  (In either case, please add a comment.)